kubespawner/spawner.py

```diff
@@ -915,22 +910,22 @@ def _expand_all(self, src):
     def _build_common_labels(self, extra_labels):
         # Default set of labels, picked up from
         # https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md
-        labels = {
-            'heritage': 'jupyterhub',
-            'app': 'jupyterhub',
-        }
-
+        labels = {}
         labels.update(extra_labels)
+        labels.update({
+            'app': os.getenv('LABEL_APP', 'jupyterhub'),
+            'release': os.getenv('LABEL_RELEASE', 'unknown')
+            'chart': os.getenv('LABEL_CHART', 'unknown')
+            'heritage': os.getenv('LABEL_HERITAGE', 'jupyterhub')
+        })
         return labels
 
     def _build_pod_labels(self, extra_labels):
-        labels = {
+        labels = self._build_common_labels(extra_labels)
+        labels.update({
             'component': 'singleuser-server'
-        }
-        labels.update(extra_labels)
-        # Make sure pod_reflector.labels in final label list
-        labels.update(self.pod_reflector.labels)
-        return self._build_common_labels(labels)
+        })
+        return labels
```

Comment:
This part contained redundant labels before, first `component` was added, then the extra labels, then `heritage` and `component` again from `PodReflector`, and finally `heritage` again and `app` from `_build_common_labels`.